lib/active_fedora/file.rb

```diff
@@ -176,6 +170,12 @@ def prefix(path)
 
     private
 
+      def create_or_update(*options)
+        super.tap do
+          metadata.save if metadata.changed?
```

Comment:
`ActiveFedora::WithMetadata::MetadataNode` does not respond to `save!` and does not have any error handling.